app/abci.go

```diff
@@ -115,6 +149,39 @@ func (app *HeimdallApp) NewProcessProposalHandler() sdk.ProcessProposalHandler {
 	return func(ctx sdk.Context, req *abci.RequestProcessProposal) (*abci.ResponseProcessProposal, error) {
 		logger := app.Logger()
 
+		var (
+			validatorSet *stakeTypes.ValidatorSet
+			err          error
+		)
+
+		chain := helper.GetConfig().Chain
+		initialHeight := int64(0)
+
+		// this is for UTs that have non zero initial height
+		if chain != helper.MainChain && chain != helper.AmoyChain && chain != helper.MumbaiChain {
```

Comment:
Same like above, also seems like a lot of code repetition, we can move it to separate function.